config/data_scrubber.go

```diff
@@ -0,0 +1,62 @@
+package config
+
+import (
+	"regexp"
+	"strings"
+)
+
+var (
+	defaultSenstiveWords = []string{"password", "passwd", "mysql_pwd", "access_token", "auth_token", "api_key", "apikey", "secret", "credentials", "stripetoken"}
+)
+
+type DataScrubber struct {
+	Enabled               bool
+	SensitiveWords        []*regexp.Regexp
+	DefaultSensitiveWords []*regexp.Regexp
+	CustomSensitiveWords  []*regexp.Regexp
+}
+
+func NewDefaultDataScrubber() *DataScrubber {
+	newDataScrubber := &DataScrubber{
+		Enabled:               true,
+		SensitiveWords:        CompileStringsToRegex(defaultSenstiveWords),
+		DefaultSensitiveWords: CompileStringsToRegex(defaultSenstiveWords),
+		CustomSensitiveWords:  make([]*regexp.Regexp, 0, 0),
+	}
+
+	return newDataScrubber
+}
+
+// Compile each word in the list into a regex pattern to match against the cmdline arguments
+func CompileStringsToRegex(words []string) []*regexp.Regexp {
+	compiledRegexps := make([]*regexp.Regexp, 0, len(words))
+	for _, word := range words {
+		pattern := `(?P<key>( |-)(?i)` + word + `)(?P<delimiter> +|=)(?P<value>[^\s]*)`
+		r := regexp.MustCompile(pattern)
```

Comment:
is it worth having better error handling than a hard panic if someone gives a word that doesn't compile as a regex? is that even possible?

Reply:
We can use Compile instead of mustCompile which sends an error when we can't parse correctly the expression. We can send the error to the NewAgentConfig function and treat it like the other errors or disable the DataScrubber by default (which can maybe confuse the user). Which option is better here?

Reply:
What about a warning when something can't compile but continue on without adding it to the list. That way an invalid regex won't block the agent running? That or just bubble up the error as you were saying.

Reply:
Yeah, that's a good point. I think you want to filter our non-alphanumeric characters from the word set so it doesn't change the behavior of the regex.

Reply:
I've done it! I've kept the test to verify it the regexp was well compiled for safe reasons even though it doesn't seem to have a problem since we only allow word characters.